src/com/tictacgo/TicTacGo.java

```diff
@@ -529,8 +530,8 @@ private void fillBoard() {
 					fl.addView(piece);
 				}
 				else { //Pieces here
-					for (int k = 0; k < board.getPieces(i, j).size(); k++) {
-						fl.addView(board.getPieces(i, j).get(k));
+					for (int k = 0; k < board.getSpace(i, j).getPieces().size(); k++) {
```

Comment:
Still a good practice to save the return value. But performance is not an issue here.